### PR TITLE
task 4

### PR DESCRIPTION
**Pull request by volha1**

### CODE REVIEW

File: task04/test/unit/helpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  test('should return true if country is correct',  () => {
+    const inputValidationResult = validateInput({country});
+    expect(inputValidationResult).toBe(true);
+  });
+
+  test('should throw error if country is invalid', () => {
+    const invalidCountry = "GB1"
+    expect(() => validateInput({country: invalidCountry})).toThrow(`Country provided is not supported, received: ${invalidCountry}`);
```
Comment (SorokinTim) [inline task04/test/unit/helpers.test.ts:21]: FYI: In practice it's better not to be tied up in the message, it's better to check the type of error being produced, but you'll be working with custom errors later, so the comment is just for information.

File: task04/test/unit/helpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import {validateInput, shortenPublicHoliday} from '../../helpers'
+import {holidays, shortHolidays} from '../test-data'
+import {SUPPORTED_COUNTRIES} from "../../config";
+
+const year = 2024;
+const country = SUPPORTED_COUNTRIES[0];
+
+describe('Validate input', () => {
```
Comment (SorokinTim) [inline task04/test/unit/helpers.test.ts]: A small note, try to use the names of services or endpoints (depending on the type of tests) in the `describe` block for better clarity, if you need more information, you can always use the nested `describe` statement.

File: task04/test/unit/public-holidays.service.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+} from '../../services/public-holidays.service';
+import {PUBLIC_HOLIDAYS_API_URL, SUPPORTED_COUNTRIES} from '../../config';
+import {holidays, shortHolidays} from '../test-data';
+
+const year = 2024;
+const country = SUPPORTED_COUNTRIES[0];
+
+describe('Get list of public holidays', () => {
+  test('should return list of short holidays',  async () => {
+    jest.spyOn(axios, 'get').mockImplementation(() => Promise.resolve({ data: holidays }));
```
Comment (SorokinTim) [inline task04/test/unit/public-holidays.service.test.ts]: I would move the axios library mock to a higher level (e.g. to the `beforeAll` level, where I would lock all necessary methods), and `mockImplementation` would already be defined at the level of specific tests.

**Review by SorokinTim — APPROVED** [review]

File: task04/test/unit/helpers.test.ts (hunk 2 above)
Comment (volha1) [reply task04/test/unit/helpers.test.ts]: Fixed

File: task04/test/unit/public-holidays.service.test.ts (hunk 3 above)
Comment (volha1) [reply task04/test/unit/public-holidays.service.test.ts]: Fixed

**Pull request merged**